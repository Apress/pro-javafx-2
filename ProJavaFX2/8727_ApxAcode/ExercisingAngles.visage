/*
 * @test
 * @run
 */

var oneDegree = 1deg;
var oneRadian = 1rad;
var oneTurn = 1turn;

println("oneDegree = {oneDegree}");
println("oneRadian = {oneRadian}");
println("oneTurn = {oneTurn}");

println("3deg + 2deg = {3deg + 2deg}");
println("5deg - 2deg = {5deg - 2deg}");
println("3deg * 2 = {3deg * 2}");
println("2 * 3deg = {2 * 3deg}");
println("2rad * 2.5 = {2rad * 2.5}");
println("2.5 * 2rad = {2.5 * 2rad}");
println("3deg / 2 = {3deg / 2}");
println("2.5rad / 2.5 = {2.5rad / 2.5}");
println("2.5deg / 5.0deg = {2.5deg / 5.0deg}");

println("oneRadian.toDegrees() = {oneRadian.toDegrees()}");
println("oneTurn.toDegrees() = {oneTurn.toDegrees()}");
println("oneDegree.toTurns() = {oneDegree.toTurns()}");

var ninetyDegrees = Angle.valueOf(10, visage.lang.AngleUnit.DEGREE);
println("ninetyDegrees = {ninetyDegrees}");
