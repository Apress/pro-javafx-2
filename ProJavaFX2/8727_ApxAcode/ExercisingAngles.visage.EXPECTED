oneDegree = 1deg
oneRadian = 1rad
oneTurn = 1turn
3deg + 2deg = 5deg
5deg - 2deg = 3deg
3deg * 2 = 6deg
2 * 3deg = 6deg
2rad * 2.5 = 5rad
2.5 * 2rad = 5rad
3deg / 2 = 1.5deg
2.5rad / 2.5 = 1rad
2.5deg / 5.0deg = 0.5
oneRadian.toDegrees() = 57.29577951308232
oneTurn.toDegrees() = 360.0
oneDegree.toTurns() = 0.0027777778
ninetyDegrees = 10deg
